From e02afe8a760733f5ae0e9f7d1d851ef19139f97a Mon Sep 17 00:00:00 2001
From: Sven Joachim <svenjoac@gmx.de>
Date: Mon, 28 Aug 2023 18:22:45 +0200
Subject: [PATCH] Create correct symlinks on non-usrmerged systems

Remove the dangling symlinks created by the 5.9 postinst first, as
otherwise "ln -s /bin/runparts /usr/bin/runparts" would fail with
EEXIST.

Closes: #1050725
---
 debian/postinst | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/debian/postinst b/debian/postinst
index 1d9f317..15b227f 100755
--- a/debian/postinst
+++ b/debian/postinst
@@ -14,7 +14,10 @@ ua() {

 usrmerge(){
 	for p in run-parts tmpfile; do
-		[ -e /usr/bin/$p ] || ln -s /bin/which /usr/bin/$p
+		[ -e /usr/bin/$p ] || rm -f /usr/bin/$p
+	done
+	for p in run-parts tempfile; do
+		[ -e /usr/bin/$p ] || ln -s /bin/$p /usr/bin/$p
 	done
 	[ -e /usr/sbin/installkernel ] || \
 				ln -s /sbin/installkernel /usr/sbin/installkernel
--
2.40.1

